nncf/tensorflow/quantization/algorithm.py

```diff
@@ -358,6 +358,8 @@ def _build_insertion_commands_for_quantizer_setup(
         non_unified_scales_quantization_point_ids = set(range(len(quantization_points)))
 
         for unified_scales_group in quantizer_setup.get_unified_scale_groups():
+            if not unified_scales_group:
```

Comment:
Could you provide justification for these changes?

Reply:
For the tensoflow precommit test `tests/tensorflow/quantization/test_unified_scales.py::test_unified_scales_with_concat` the error occurred on the line due to an empty group.https://github.com/openvinotoolkit/nncf/blob/0e90dac127b08dee763aee47020211b73a5d1ddd/nncf/tensorflow/quantization/algorithm.py#L361

The group becomes empty because all its members were removed by new filer pass. This happens for a synthetic model from a function, which has multiply operation (elementwise) with two same inputs, which is a very synthetic case in my opinion. 
https://github.com/openvinotoolkit/nncf/blob/0e90dac127b08dee763aee47020211b73a5d1ddd/tests/tensorflow/quantization/test_unified_scales.py#L24

So, two quantizers for multiply (both placed on the same input) were unified and then removed by the new proposed logic (because these 2 edges represented as 1), but the group of quantizers remained. 
These changes aim to fix this issue when a group is empty, which is possible. 
Another possible option is to make this synthetic model more realistic (without sharing inputs).